scripts/add_action_doc_to_proto.awk

```diff
@@ -0,0 +1,69 @@
+BEGIN {
```

Comment:
Who doesn't love a good awk script? I have documented the key points, but assumes some basic familiarity with awk.

Reply:
Is there an advantage to using awk rather than something like ruby that more folks at the company will likely feel comfortable working with?

Reply:
Just saw your comment after I merged. Probably no advantage, no. Given the linux expertise in the company I reasoned that awk would be an acceptable choice; I did not really consider ruby because I have not personally used it much.